src/base/bittorrent/session.cpp

```diff
@@ -2349,10 +2349,10 @@ void Session::exportTorrentFile(TorrentHandle *const torrent, TorrentExportFolde
 void Session::generateResumeData(bool final)
 {
     foreach (TorrentHandle *const torrent, m_torrents) {
-        if (!torrent->isValid()) continue;
-        if (torrent->hasMissingFiles()) continue;
-        if (torrent->isChecking() || torrent->hasError()) continue;
+        if (torrent->isChecking() || torrent->isPaused()) continue;
         if (!final && !torrent->needSaveResumeData()) continue;
+        if (torrent->hasMissingFiles() || torrent->hasError()) continue;
```

Comment:
Please move these lines back to make the change set as minimal as possible, unless the reordering is required, in which case it should be reflected in the commit message.

Reply:
Ok!

Reply:
Thanks.